Source/Core/VideoBackends/Metal/MTLUtil.mm

```diff
@@ -0,0 +1,470 @@
+{
+  config->backend_info.api_type = APIType::Metal;
+  config->backend_info.bUsesLowerLeftOrigin = false;
+  config->backend_info.bSupportsExclusiveFullscreen = false;
+  config->backend_info.bSupportsDualSourceBlend = true;
+  config->backend_info.bSupportsPrimitiveRestart = true;
+  config->backend_info.bSupportsGeometryShaders = false;
+  config->backend_info.bSupportsComputeShaders = true;
+  config->backend_info.bSupports3DVision = false;
+  config->backend_info.bSupportsEarlyZ = true;
+  config->backend_info.bSupportsBindingLayout = true;
+  config->backend_info.bSupportsBBox = true;
+  config->backend_info.bSupportsGSInstancing = false;
+  config->backend_info.bSupportsPostProcessing = true;
+  config->backend_info.bSupportsPaletteConversion = true;
+  config->backend_info.bSupportsClipControl = true;
+  config->backend_info.bSupportsSSAA = true;
+  config->backend_info.bSupportsFragmentStoresAndAtomics = true;
+  config->backend_info.bSupportsReversedDepthRange = false;
+  config->backend_info.bSupportsLogicOp = false;
+  config->backend_info.bSupportsMultithreading = false;
+  config->backend_info.bSupportsGPUTextureDecoding = true;
+  config->backend_info.bSupportsCopyToVram = true;
+  config->backend_info.bSupportsBitfield = true;
+  config->backend_info.bSupportsDynamicSamplerIndexing = true;
+  config->backend_info.bSupportsFramebufferFetch = false;
+  config->backend_info.bSupportsBackgroundCompiling = true;
+  config->backend_info.bSupportsLargePoints = true;
+  config->backend_info.bSupportsPartialDepthCopies = true;
+  config->backend_info.bSupportsDepthReadback = true;
+  config->backend_info.bSupportsShaderBinaries = false;
+  config->backend_info.bSupportsPipelineCacheData = false;
+  config->backend_info.bSupportsCoarseDerivatives = false;
+  config->backend_info.bSupportsTextureQueryLevels = true;
+  config->backend_info.bSupportsLodBiasInSampler = false;
+  config->backend_info.bSupportsSettingObjectNames = true;
+  // Metal requires multisample resolve to be done on a render pass
+  config->backend_info.bSupportsPartialMultisampleResolve = false;
+}
+
+void Metal::Util::PopulateBackendInfoAdapters(VideoConfig* config,
+                                              const std::vector<MRCOwned<id<MTLDevice>>>& adapters)
+{
+  config->backend_info.Adapters.clear();
+  for (id<MTLDevice> adapter : adapters)
+  {
+    config->backend_info.Adapters.push_back([[adapter name] UTF8String]);
+  }
+}
+
+static bool DetectIntelGPUFBFetch(id<MTLDevice> dev)
+{
+  // Even though it's nowhere in the feature set tables, some Intel GPUs support fbfetch!
+  // Annoyingly, the Haswell compiler successfully makes a pipeline but actually miscompiles it and
+  // doesn't insert any fbfetch instructions The Broadwell compiler inserts the Skylake fbfetch
+  // instruction, but Broadwell doesn't support that.  It seems to make the shader not do anything
+  // So we actually have to test the thing
```

Comment:
It looks like this comment got re-wrapped, when there should be separate lines between "fbfetch instructions" and "The Broadwell". The formatter does that when it reflows comments that are too long (I think), but it won't do it if the comment stays below the line-length limits.  (IMO it would also be useful to add periods at the end of sentences here - I personally am inconsistent with that, but since there are multiple sentences it seems like a good idea in this case.)